src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -98,6 +98,7 @@ enum class HintType(desc: String, enabled: Boolean) {
             val element = elem as? RsLambdaExpr ?: return emptyList()
             val type = element.type as? TyFunction ?: return emptyList()
             return element.valueParameterList.valueParameterList
+                .filter { it.typeReference == null }
```

Comment:
These changes don't take into account cases when lambda has several arguments and only some of them have explicit type parameters
For example:
```rust
struct S;
fn foo<T: Fn(S, S, (S, S)) -> ()>(action: T) {}
fn main() {
    foo(|x, y: S, z| {});
}
```
Here type hint for `z` is wrong
<img width="329" alt="screen shot 2018-01-30 at 11 32 21" src="https://user-images.githubusercontent.com/2539310/35555784-57e3d5ee-05b1-11e8-8c8e-3c72df9a73f5.png">